test/functional/neutronless/test_disconnected_service.py

```diff
@@ -0,0 +1,149 @@
+DISCONNECTED_SVC['listeners'][0]['id'] = LISTENER_ID
+CONNECTED_SVC['pools'][0]['listeners'][0]['id'] = LISTENER_ID
+CONNECTED_SVC['listeners'][0]['id'] = LISTENER_ID
+for listener in DISCONNECTED_SVC['listeners']:
+    listener['provisioning_status'] = 'PENDING_CREATE'
+for pool in DISCONNECTED_SVC['pools']:
+    pool['provisioning_status'] = 'PENDING_CREATE'
+for monitor in DISCONNECTED_SVC['healthmonitors']:
+    monitor['provisioning_status'] = 'PENDING_CREATE'
+for member in DISCONNECTED_SVC['members']:
+    member['provisioning_status'] = 'PENDING_CREATE'
+    member['network_id'] = u'a8f301b2-b7b9-404a-a746-53c442fa1a4f'
+
+for listener in CONNECTED_SVC['listeners']:
+    listener['provisioning_status'] = 'PENDING_CREATE'
+for pool in CONNECTED_SVC['pools']:
+    pool['provisioning_status'] = 'PENDING_CREATE'
+for monitor in CONNECTED_SVC['healthmonitors']:
+    monitor['provisioning_status'] = 'PENDING_CREATE'
+for member in CONNECTED_SVC['members']:
+    member['provisioning_status'] = 'PENDING_CREATE'
+    member['network_id'] = u'a8f301b2-b7b9-404a-a746-53c442fa1a4f'
+
+
+@pytest.fixture(scope='module')
+def setup_registry_snapshot(bigip):
+    # Setup device registries
+    pp('inside setup_registry_snapshot')
+    return frozenset(register_device(bigip))
+
+
+@pytest.fixture
+def setup_neutronless(request, bigip, setup_registry_snapshot):
+    pretest_snapshot = setup_registry_snapshot
+    """F5 LBaaS agent for OpenStack."""
+    # Setup neutronless icontroldriver
+    for element, value in symbols.__dict__.items():
+        setattr(cfg.CONF, element, value)
+    icontroldriver = iControlDriver(cfg.CONF, registerOpts=False)
+    icontroldriver.plugin_rpc = mock.MagicMock()
+    wrappedicontroldriver = mock.MagicMock(wraps=icontroldriver)
+
+    def _deletion_order(to_delete):
+        ordering = {'/mgmt/tm/ltm/virtual': 1,
+                    '/mgmt/tm/ltm/pool': 2,
+                    'mgmt/tm/ltm/node/': 3,
+                    'monitor': 4,
+                    'virtual-address': 5,
+                    '/mgmt/tm/net/fdb/tunnel': 6,
+                    'mgmt/tm/net/tunnels/tunnel/': 7,
+                    '/mgmt/tm/sys/folder': 8}
+
+        def order_key(item):
+            for k in ordering:
+                if k in item:
+                    return ordering[k]
+            return 999
+        ordered_for_deletion = sorted(list(to_delete), key=order_key)
+        return ordered_for_deletion
+
+    def remove_test_created_elements():
+        posttest_registry = register_device(bigip)
+        created = frozenset(posttest_registry) - pretest_snapshot
+        ordered = _deletion_order(created)
+        for selfLink in ordered:
+            if 'virtual-address' not in selfLink:
+                posttest_registry[selfLink].delete()
+
+    request.addfinalizer(remove_test_created_elements)
+    return bigip, wrappedicontroldriver, pretest_snapshot
+
+
+def test_disconnected_service(setup_neutronless):
```

Comment:
This replaces the functional test I wrote that requires a live deployment.  There are many additional tests required to qualify the feature.

- create/delete of the 'disconnected network' (i.e. ppp interface) upon tenant folder creation/deletion
- 3 new ini settings (at least the two for polling interval and gross timeout)
- "edge" tests such as a service containing the seg_id immediately upon creation